src/io/write-diary.cpp

```diff
@@ -271,9 +268,7 @@ errr exe_write_diary(PlayerType *player_ptr, int type, int num, concptr note)
         return -1;
     }
 
-    concptr note_level = "";
-    char note_level_buf[40];
-    auto q_idx = write_floor(player_ptr, &note_level, note_level_buf);
+    auto floor_result = write_floor(player_ptr);
```

Comment:
Since floor_result.first and floor_result.second are used frequently after this, it would be better to receive them in separate variables in a structured binding as follows, and use them instead of floor_result.first and floor_result.second.

```c++
const auto [q_idx, note_level] = write_floor(player_ptr);
```

Reply:
The forced push makes that substitution and gets rid of the excess append() operations as noted in the feedback on the other pull requests.